blockchain/blockchain.go

```diff
@@ -208,3 +208,134 @@ func getBlockByHash(txn db.Transaction, hash *felt.Felt) (*core.Block, error) {
 		return getBlockByNumberBytes(txn, numBytes)
 	}
 }
+
+// GetTransactionByBlockNumAndIndex gets the transaction for a given block number and index.
+func (b *Blockchain) GetTransactionByBlockNumAndIndex(blockNumber, index uint64) (transaction core.Transaction, err error) {
+	bnIndexBytes := joinBlockNumberAndIndex(blockNumber, index)
+	return transaction, b.database.View(func(txn db.Transaction) error {
+		transaction, err = getTransactionByBnIndexBytes(txn, bnIndexBytes)
+		return err
+	})
+}
+
+// GetTransactionByHash gets the transaction for a given hash.
+func (b *Blockchain) GetTransactionByHash(hash *felt.Felt) (transaction core.Transaction, err error) {
+	return transaction, b.database.View(func(txn db.Transaction) error {
+		transaction, err = getTransactionByHash(txn, hash)
+		return err
+	})
+}
+
+// StoreTransaction takes a transaction and verifies its hash before storing it in the database.
+func (b *Blockchain) StoreTransaction(blockNumber, txnIndex uint64, transaction core.Transaction) error {
```

Comment:
`StoreTransaction` and `StoreReceipt` should not be public. They should be folded in to `StoreBlock`

Reply:
What about tests?

Reply:
There should be a public function called GetTransaction/GetTransactionReceipt. So after storing you call that to see that you get the expected output

Reply:
[Here](https://github.com/NethermindEth/juno/pull/534/files#diff-37b6e7ed4f299b87b357b54376123393b34ee0699fc035f25bc0621627dcc6baR221-R235)